### PR TITLE
Create a labels controller and shared views

### PR DESCRIPTION
**Pull request by hellcp-work**
Implements a controller and base shared views for future implementation in package (https://github.com/openSUSE/open-build-service/pull/16630) and request views

### CODE REVIEW

File: src/api/app/views/webui/shared/_label_form.html.haml [hunk 1]
```diff
@@ -2,15 +2,17 @@
...
-    = form_for :labels, url: labels_path(project:), method: :put do |f|
-      = f.hidden_field :labelable_type, value: labelable.class
-      = f.hidden_field :labelable_id, value: labelable.id
-      = f.collection_check_boxes(:label_template_ids, project.label_templates, :id, :name) do |l|
-        .dropdown-item
-          .form-check
-            = l.check_box class: 'form-check-input', checked: existing_labels.include?(l.object.id)
-            = l.label class: 'form-check-label' do
-              = render partial: 'webui/shared/label', locals: { label: l.object }
+    - existing_labels = labelable.labels.to_h { |l| [l.label_template_id, l.id] }
```
Comment (hennevogel) [inline src/api/app/views/webui/shared/_label_form.html.haml:5]:
```suggestion
    - existing_labels = labelable.labels.map { |l| [l.label_template_id, l.id] }
```

Comment (hellcp-work) [reply src/api/app/views/webui/shared/_label_form.html.haml:5]: Those two do different things though, we would have to do `map {...}.to_h` if we want to use `map`

File: src/api/app/views/webui/shared/_label_form.html.haml [hunk 2]
```diff
@@ -2,15 +2,17 @@
...
+    = form_for :labels, url: labels_path(project:, labelable_id: labelable.id, labelable_type: labelable.class), method: :put do |f|
+      - project.label_templates.each do |label_template|
+        = f.fields_for 'labels_attributes[]', label_template do |la|
+          .dropdown-item
+            = la.hidden_field :label_template_id, value: la.object.id
+            - if existing_labels.keys.include?(la.object.id)
```
[the remark this answers is not in the document]
Comment (hennevogel) [reply src/api/app/views/webui/shared/_label_form.html.haml:11]: then you don't need to build a hash...

[the remark this answers is not in the document]
Comment (hellcp-work) [reply src/api/app/views/webui/shared/_label_form.html.haml:11]: I assume by that you mean the following:
```suggestion
            - if labelable.labels.exists?(label_template: la.object)
```

File: src/api/spec/controllers/webui/labels_controller_spec.rb [hunk 3]
```diff
@@ -0,0 +1,30 @@
...
+      before do
+        put :update,
+            params: { project: home_tom.name, labelable_id: toms_package.id, labelable_type: 'Package',
+                      labels: { labels_attributes: [{ label_template_id: label_one.id, _destroy: false },
+                                                    { label_template_id: label_two.id, _destroy: true },
+                                                    { label_template_id: label_three.id, _destroy: false }] } }
+      end
+
+      it 'creates two labels' do
+        expect(toms_package.labels.pluck(:label_template_id)).to eq([label_one.id, label_three.id])
```
Comment (hennevogel) [inline src/api/spec/controllers/webui/labels_controller_spec.rb:26]: What are you testing here?

Comment (hellcp-work) [reply src/api/spec/controllers/webui/labels_controller_spec.rb:26]: If the labels get applied after submitting the form with the right values

File: src/api/app/controllers/webui/labels_controller.rb [hunk 4]
```diff
@@ -0,0 +1,29 @@
+class Webui::LabelsController < Webui::WebuiController
+  before_action :set_project
+  before_action :set_labelable
+
+  def update
+    authorize @labelable, :update?
```
Comment (danidoni) [inline src/api/app/controllers/webui/labels_controller.rb]: The `:update?` is not really needed, Pundit will detect the controller action and call the appropriate policy method

Comment (hellcp-work) [reply src/api/app/controllers/webui/labels_controller.rb]: This is now `update_labels?`, which is not the same as the controller action

File: src/api/app/policies/package_policy.rb [hunk 5]
```diff
@@ -22,6 +22,10 @@ def update?
     user.can_modify?(record)
   end
 
+  def update_labels?
```
Comment (danidoni) [inline src/api/app/policies/package_policy.rb:25]: Why having a different policy method?

Comment (hellcp-work) [reply src/api/app/policies/package_policy.rb:25]: Because of BsRequests, which aren't covered by this PR

Comment (danidoni) [reply src/api/app/policies/package_policy.rb:25]: Alright...

**Comment by danidoni** [comment]
So, is this not used anywhere?

**Comment by hellcp-work** [comment]
There's a PR implementing it for packages #16630

**Review by danidoni — APPROVED** [review]

**Pull request merged**